server/Dockerfile

```diff
@@ -0,0 +1,60 @@
+# create base image with all packages, proper entrypoint, and directories created
+FROM node:erbium-alpine AS base
+
+# install any packages we need from apt here
+RUN apk add --update dumb-init
+
+# set entrypoint to `dumb-init` as it handles being pid 1 and forwarding signals
+# so that you dont need to bake that logic into your node app
+ENTRYPOINT ["dumb-init", "--"]
+
+# all of our code will live in `/app`
+WORKDIR /app
+
+# using the base image, create an image containing all of our files
+# and dependencies installed, devDeps and test directory included
+FROM base AS dependencies
+
+COPY package*.json ./
+RUN npm set progress=false \
+    && npm config set depth 0 \
+    && npm i
+
+COPY .eslintrc.json .
+COPY ./server ./server
+COPY ./test ./test
+COPY ./lib ./lib
+
+# `node` user is created in base node image
+# we want to use this non-root user for running the server in case of attack
+# https://github.com/nodejs/docker-node/blob/master/docs/BestPractices.md#non-root-user
+USER node
+
+# if you have any build scripts to run, like for the `templated-site` flavor
+# uncomment and possibly modify the following RUN command:
+# RUN npm run build
+# keeping all of the bash commands you can within a single RUN is generally important,
+# but for this case it's likely that we want to use the cache from the prune which will
+# change infrequently.
+
+# test running image using all of the files and devDeps
+FROM dependencies AS test
+ENV NODE_ENV=test
+# use `sh -c` so we can chain test commands using `&&`
+CMD ["npm", "test"]
+
+# dev ready image, contains devDeps for test running and debugging
+FROM dependencies AS development
+
+# expose port 3000 from in the container to the outside world
+# this is the default port defined in server/manifest.js, and
+# will need to be updated if you change the default port
+EXPOSE 3000
+CMD ["npm", "start"]
+
+# release ready image, devDeps are pruned and tests removed for size control
+FROM development AS release
+
+# prune non-prod dependencies, remove test files
+RUN npm prune --production \
```

Comment:
I am still having perms trouble with this that I can fix by switching to root before running this command then back to node:
```
$ docker build --target release --file server/Dockerfile .
Step 1/20 : FROM node:erbium-alpine AS base
erbium-alpine: Pulling from library/node
Digest: sha256:bba77d0ca8820b43af898b3c50d4e8b68dc703ebbd958319af2f21f2d3c309f5
Status: Downloaded newer image for node:erbium-alpine
 ---> 927d03058714
Step 2/20 : RUN apk add --update dumb-init
...
...
Step 19/20 : FROM development AS release
 ---> dcc6968ead14
Step 20/20 : RUN npm prune --production     && rm -rf ./test
 ---> Running in 58d38382fb64
npm WARN checkPermissions Missing write access to node_modules/@babel/generator/node_modules/source-map
npm WARN checkPermissions Missing write access to node_modules/@babel/helper-validator-identifier
npm WARN checkPermissions Missing write access to node_modules/@babel/parser
npm WARN checkPermissions Missing write access to node_modules/@hapi/bossy
npm WARN checkPermissions Missing write access to node_modules/@hapi/rule-capitalize-modules
... (many more similar errors in node_modules, apparently all the dev deps)
npm ERR! code EACCES
npm ERR! syscall access
npm ERR! path node_modules/@babel/generator/node_modules/source-map
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, access 'node_modules/@babel/generator/node_modules/source-map'
npm ERR!  [Error: EACCES: permission denied, access 'node_modules/@babel/generator/node_modules/source-map'] {
npm ERR!   stack: "Error: EACCES: permission denied, access 'node_modules/@babel/generator/node_modules/source-map'",
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: 'node_modules/@babel/generator/node_modules/source-map'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2020-03-22T17_34_02_879Z-debug.log
The command '/bin/sh -c npm prune --production     && rm -rf ./test' returned a non-zero code: 243
```